### PR TITLE
WIP: Feature/update ts transform plugin

### PR DESCRIPTION
**Pull request by thekip**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:https://github.com/nestjs/graphql/issues/1346

## What is the new behavior?
This PR brings changes in how current ts transform plugin is working. 

To Do:
- [x] Option to work in isolatedModules (no typeChecker available)
- [x] Introspection for `@deprecated` jsdoc tag
- [ ] Support for GraphQL scalar types: Int, Float, ID (please refer to the issue https://github.com/nestjs/graphql/issues/1346 i rised a discussion there)
- [ ] Support more complicated cases in no-typechecker mode.
- [ ] More tests covering edge cases in no-typechecker mode. 

## What was changed internally: 
1. JSDoc Comments now parsed using Typescript AST (built-it typescript support) instead of regular expressions 
2. Removed code related to imports and merging existing factories (please see linked issue)
3.  Test rewritten, Jest Snapshots used

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```
There is a defenetely changes which can impact users. 
TODO: eliminate them or at least find and clearly state what can go wrong. 

## Other information

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -9,7 +9,6 @@ module.exports = {
     'plugin:@typescript-eslint/eslint-recommended',
     'plugin:@typescript-eslint/recommended',
     'prettier',
-    'prettier/@typescript-eslint',
```
Comment (thekip) [inline .eslintrc.js:12]: https://stackoverflow.com/a/66342626/2210610

File: lib/plugin/merge-options.ts [hunk 2]
```diff
@@ -3,11 +3,13 @@ import { isString } from '@nestjs/common/utils/shared.utils';
 export interface PluginOptions {
   typeFileNameSuffix?: string | string[];
   introspectComments?: boolean;
+  advancedTypeIntrospection?: boolean;
 }
 
 const defaultOptions: PluginOptions = {
   typeFileNameSuffix: ['.input.ts', '.args.ts', '.entity.ts', '.model.ts'],
   introspectComments: false,
+  advancedTypeIntrospection: true,
```
Comment (thekip) [inline lib/plugin/merge-options.ts:12]: this flag enables "old" behavior and aimed to eliminate most of breaking changes.
